### ISSUE TITLE
[BUG] The specified configuration does not have a telemetry channel. (Parameter 'configuration')

### ISSUE THREAD

**Issue by algattik** [Azure/azure-sdk-for-net#19068]
**Describe the bug**
When creating SDK clients in parallel in integration tests, each test recreates an instance of DefaultAzureCredential grabbing an Azure CLI credential. In a few percent of the cases, this fails with an exception.

**Actual behavior (include Exception or Stack Trace)**

```text
System.ArgumentException: The specified configuration does not have a telemetry channel. (Parameter 'configuration')
at Microsoft.ApplicationInsights.TelemetryClient..ctor(TelemetryConfiguration configuration)
at Microsoft.ApplicationInsights.DependencyCollector.Implementation.EventHandlers.DiagnosticsEventHandlerBase..ctor(TelemetryConfiguration configuration)
at Microsoft.ApplicationInsights.DependencyCollector.Implementation.AzureSdkDiagnosticsEventHandler..ctor(TelemetryConfiguration configuration)
at Microsoft.ApplicationInsights.DependencyCollector.Implementation.AzureSdkDiagnosticListenerSubscriber.GetEventHandler(String diagnosticListenerName)
at Microsoft.ApplicationInsights.DependencyCollector.Implementation.DiagnosticSourceListenerBase`1.OnNext(DiagnosticListener value)
at System.Diagnostics.DiagnosticListener.AllListenerObservable.OnNewDiagnosticListener(DiagnosticListener diagnosticListener)
at System.Diagnostics.DiagnosticListener..ctor(String name)
at Azure.Core.Pipeline.DiagnosticScopeFactory..ctor(String clientNamespace, String resourceProviderNamespace, Boolean isActivityEnabled)
at Azure.Core.Pipeline.ClientDiagnostics..ctor(ClientOptions options)
at Azure.Identity.CredentialPipeline..ctor(TokenCredentialOptions options)
at Azure.Identity.CredentialPipeline.GetInstance(TokenCredentialOptions options)
at Azure.Identity.DefaultAzureCredentialFactory..ctor(TokenCredentialOptions options)
at Azure.Identity.DefaultAzureCredential..ctor(DefaultAzureCredentialOptions options)
```

**To Reproduce**
Steps to reproduce the behavior (include a code snippet, screenshot, or any additional information that might help us reproduce the issue)

```csharp
            services.AddApplicationInsightsTelemetry(o =>
            {
                o.RequestCollectionOptions.TrackExceptions = true;
            })
                .AddSingleton<TokenCredential>(s =>
                    new DefaultAzureCredential(new DefaultAzureCredentialOptions
                    {
                        ExcludeVisualStudioCodeCredential = true,
                        ExcludeSharedTokenCacheCredential = true,
                        ExcludeInteractiveBrowserCredential = true,
                        ExcludeEnvironmentCredential = true,
                        ExcludeVisualStudioCredential = true,
                        ExcludeManagedIdentityCredential = true,
                    }))
                .AddSingleton(s => new DigitalTwinsClient(new Uri(Configuration["ADT:InstanceUrl"]), s.GetRequiredService<TokenCredential>()))
              
```

Then create several integration tests (with `IClassFixture<WebApplicationFactory<Startup>>`)

**Environment:**

- "Azure.DigitalTwins.Core" Version="1.2.0"; "Azure.Identity" Version="1.3.0" 
- dotnet 3.1

.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.101/

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  2.1.811 [/usr/local/share/dotnet/sdk]
  3.1.404 [/usr/local/share/dotnet/sdk]
  5.0.101 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.23 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.23 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.23 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

could also reproduce on Azure Devops Ubuntu agent.

**Comment by jsquire** [comment]
Hi @algattik:   Is there a reason why you're using `DefaultAzureCredential` in the tests rather than the [`Credential`](https://github.com/Azure/azure-sdk-for-net/blob/3a29afb7ba28e3bc7d8a15ffadecb5f4ee363889/sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs#L156) member that your `TestEnvironment` instance offers?

**Comment by algattik** [comment]
Hi @jsquire, at the moment we use the same Startup class used in the runtime code, but I could try that.

**Comment by algattik** [comment]
`TestEnvironment` seems to be part of azure-sdk-for-net's internal test framework. We are building an enterprise project and need to work with standard `WebApplicationFactory`. Otherwise please advise on using `TestEnvironment`  externally.

Also note that we use AzureCliCredential for developer convenience, which apparently `TestEnvironment` doesn't support.

**Comment by jsquire** [comment]
@algattik:  Apologies; I overlooked that "customer" part of the text in your contact card and assumed based on your Azure membership that the integration tests that you referred to were part of the SDK itself.  You're correct in that the `TestEnvironment` isn't the way to go here.   I'll need to defer to @christothes and @schaabs as the Identity experts for better guidance.

**Comment by christothes** [comment]
Hi @algattik This appears to be an issue with Application Insights. Would you mind opening an issue here - https://github.com/microsoft/ApplicationInsights-dotnet ?

**Comment by nulltoken** [comment]
@christothes @jsquire 

:wave: I'm also seeing such outcome, but with a different call stack. Creating a repro case in my context may actually be non trivial.

This issue has been closed and I'm a little lost as to how this issue should be handled.

Any guidance would be greatly appreciated. Thoughts?

References:
- Azure.Storage.Blobs: 12.8.0
- Microsoft.ApplicationInsights.AspNetCore: 2.17.0

```
  Message: 
    System.ArgumentException : The specified configuration does not have a telemetry channel. (Parameter 'configuration')
  Stack Trace: 
    TelemetryClient.ctor(TelemetryConfiguration configuration)
    DiagnosticsEventHandlerBase.ctor(TelemetryConfiguration configuration)
    AzureSdkDiagnosticsEventHandler.ctor(TelemetryConfiguration configuration)
    AzureSdkDiagnosticListenerSubscriber.GetEventHandler(String diagnosticListenerName)
    DiagnosticSourceListenerBase`1.OnNext(DiagnosticListener value)
    AllListenerObservable.OnNewDiagnosticListener(DiagnosticListener diagnosticListener)
    DiagnosticListener.ctor(String name)
    DiagnosticScopeFactory.ctor(String clientNamespace, String resourceProviderNamespace, Boolean isActivityEnabled)
    ClientDiagnostics.ctor(ClientOptions options)
    StorageRequestValidationPipelinePolicy.ctor(ClientOptions options)
    StorageClientOptions.Build(ClientOptions options, HttpPipelinePolicy authentication, Uri geoRedundantSecondaryStorageUri)
    StorageClientOptions.Build(ClientOptions options, Object credentials, Uri geoRedundantSecondaryStorageUri)
    BlobClientOptions.Build(Object credentials)
    BlobContainerClient.ctor(String connectionString, String blobContainerName, BlobClientOptions options)
    BlobContainerClient.ctor(String connectionString, String blobContainerName)
    BlobsDownloader.BuildBlobContainerAccessor(IConfiguration configuration, String connectionStringKey, String name) line 96
    PublicFilesFixture.ctor(PublicFilesWebServerSetupFixture wss) line 27
```

**Comment by jsquire** [comment]
Hi @nulltoken.   I'd advise opening a new issue for the Application Insights team, as this looks like it is most likely an issue in that library.  https://github.com/microsoft/ApplicationInsights-dotnet

**Comment by nulltoken** [comment]
@jsquire Thanks. I will do this.
